contents/ruby/Rails.md

```diff
@@ -0,0 +1,157 @@
+               edit_entry GET    /entries/:id/edit(.:format)                                                              entries#edit
+                    entry GET    /entries/:id(.:format)                                                                   entries#show
+                          PATCH  /entries/:id(.:format)                                                                   entries#update
+                          PUT    /entries/:id(.:format)                                                                   entries#update
+                          DELETE /entries/:id(.:format)                                                                   entries#destroy
+``` 
+
+The word 'automagically' should finally start to make sense at this point! Instead of going through the arduous process of writing out the code dump at the top, and deciding the specific names of the routes, as well as the corresponding functions in the controller, Rails has already configured all the URI patterns with function names in the controller for entries. Now all you need to do is focus on writing the function logic, instead of coding out the names of the individual URIs and routes.
+
+Many other trivial tasks that require boilerplate code are similarly done away with in Rails, thus making developement a more enjoyable (more on that in a bit), and a much faster process. All of these optimizations help make developing an application between 30-40% faster on Rails, while also making it much easier to learn as compared to competing frameworks.  
+  
+### Optimised for Programmer Happiness
+
+One of the primary tenets of Ruby was to make it the 'Least Surprising Language', where the language was designed to perform exactly as the developer thought would be the most 'expected' function. Rails founder DHH built on that, and optimized Rails for Programmer Happiness. This meant re-imagining many of coding's most standard conventions, as well as adding significant bloat, all for the purpose of making the code look 'pretty', and inherently 'nice' to write (as subjective as that sounds). 
+
+For example, in order to fetch yesterday's date in Python, one may have to write
+
+```python
+import datetime
+datetime.datetime.now() - datetime.timedelta(days = 1)
+```
+
+The same action in Rails is done using
+```ruby
+1.day.ago
+```
+
+Similarly, Rails also added a new method to access an array element. Instead of following the C convention of using square-brackets to access the second element in the array 'elements' like
+
+```c
+elements[1]
+```
+
+Rails developers can use ordinal representation to access the same element with the command
+
+```ruby
+elements.second
+```
+
+And while this may not reduce the amount one has to type, most Rails enthusiasts agree that it makes the code much more readable, intuitive, and inherently happiness-inducing.
+
+### Great Community of Developers
```

Comment:
What a less subjective headline might look like:
> Large and Active Community of Developers

Reply:
I'm fine with `Large and Active Community of Developers` :)